packages/smart-contracts/src/contracts/EthereumSingleRequestProxy.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import './interfaces/EthereumFeeProxy.sol';
+
+/**
+ * @title EthereumSingleRequestProxy
+ * @notice This contract is used to send a single request to a payee with a fee sent to a third address
+ */
+contract EthereumSingleRequestProxy {
+  address public payee;
+  bytes public paymentReference;
+  address public feeAddress;
+  uint256 public feeAmount;
+  IEthereumFeeProxy public ethereumFeeProxy;
+
+  address private originalSender;
+
+  // Reentrancy guard
+  uint256 private constant _NOT_ENTERED = 1;
+  uint256 private constant _ENTERED = 2;
+  uint256 private _status;
```

Comment:
We could use openzepplin `ReentrancyGuard` contract instead

Reply:
This is the same implementation as openzeppelin `ReentrancyGuard` with an edited modifier to allow reentry to the `EthereumFeeProxy` while still checking other users.